armi/reactor/geometry.py

```diff
@@ -113,6 +117,269 @@ def __str__(self):
+            return 1.0
+        elif self == self.THIRD_CORE:
+            return 3.0
+        elif self == self.QUARTER_CORE:
+            return 4.0
+        elif self == self.EIGHTH_CORE:
+            return 8.0
+        elif self == self.SIXTEENTH_CORE:
+            return 16.0
+        else:
+            raise ValueError(
+                "Could not calculate symmetry factor for shape {}. update logic.".format(
+                    self.label
+                )
+            )
+
+
+class BoundaryType(enum.Enum):
+    """
+    Enumeration of boundary types.
+    """
+
+    NO_SYMMETRY = 0
+    PERIODIC = 1
+    REFLECTIVE = 2
+
+    @classmethod
+    def fromAny(cls, source: Union[str, "BoundaryType"]) -> "BoundaryType":
+        """
+        Safely convert from string representation, no-op if already an enum instance.
+
+        This is useful as we transition to using enumerations more throughout the code.
+        There will remain situations where a geomType may be provided in string or enum
+        form, in which the consuming code would have to check the type before
+        proceeding. This function serves two useful purposes:
+         - Relieve client code from having to if/elif/else on ``isinstance()`` checks
+         - Provide a location to instrument these conversions for when we actually try
+           to deprecate the strings. E.g., produce a warning when this is called, or
+           eventually forbidding the conversion entirely.
+        """
+        if isinstance(source, BoundaryType):
+            return source
+        elif isinstance(source, str):
+            return cls.fromStr(source)
+        else:
+            raise TypeError("Expected str or BoundaryType; got {}".format(type(source)))
+
+    @classmethod
+    def fromStr(cls, symmetryStr: str) -> "BoundaryType":
+        # case-insensitive
+        x = symmetryStr.lower().strip()
+        for canonical in (x, parsing.findString(boundaryTypes, x)):
+            if canonical == NO_SYMMETRY:
+                return cls.NO_SYMMETRY
+            elif canonical == PERIODIC:
+                return cls.PERIODIC
+            elif canonical == REFLECTIVE:
+                return cls.REFLECTIVE
+        return cls.NO_SYMMETRY
+
+    @property
+    def label(self):
+        """Human-presentable label"""
+
+        if self == self.NO_SYMMETRY:
+            return "No Symmetry"
+        elif self == self.REFLECTIVE:
+            return "Reflective"
+        elif self == self.PERIODIC:
+            return "Periodic"
+
+    def __str__(self):
+        """Inverse of fromStr()"""
+        if self == self.NO_SYMMETRY:
+            return NO_SYMMETRY  # should we return an empty string here instead?
+        elif self == self.PERIODIC:
+            return PERIODIC
+        elif self == self.REFLECTIVE:
+            return REFLECTIVE
+
+    def hasSymmetry(self):
+        return not self == self.NO_SYMMETRY
+
+
+def _joinSpace(strList: List[str]):
+    return " ".join(strList)
+
```

Comment:
No need to replace the one-liner with a function; just use `" ".join()` in situ or make a function that renders the proper string for the combination of geom type and symmetry.

Reply:
Yeah, having flipped through where it's called i think something like `def geomDeclaration(geomType: GeomType, symmetryType: SymmetryType) -> str:` would make more sense, since it could even handle the through-center behavior properly with less fuss.

Reply:
We actually rarely need to get a string representation of the `geomType` and `symmetryType` combined, so I didn't implement this. I just used the `SymmetryType ` representation in most places.